Erratic layout change on desktop switch

# Issue description

I have experimented the same issue described in #1945 but I guess this is more general trouble around  EWMH/NetWM interactions. If I use [wmctrl](http://tripie.sweb.cz/utils/wmctrl/) to try to interact with qtile, it regularly causes layout change on current desktop.

More precisely, if I try to switch desktop, it switched layout and not desktop : `wmctrl -s 1` don't navigate to desktop 1 but change to the second layout available (index 1). 

# Qtile version

Latest stable, from Archlinux repository.

I have not included stack trace nor my configuration as I guess it's not relevant here (multiple groups and layouts activated, no errors) but please let me know if I'm wrong.

That said, if I use `wmctrl -s 9` for example, I got a trace

```
2020-11-19 15:56:07,284 ERROR libqtile xcore.py:_xpoll():L277 Got an exception in poll loop
Traceback (most recent call last):
  File "/usr/lib/python3.8/site-packages/libqtile/backend/x11/xcore.py", line 245, in _xpoll
    ret = target(event)
  File "/usr/lib/python3.8/site-packages/libqtile/backend/x11/xcore.py", line 528, in handle_ClientMessage
    self.qtile.cmd_to_layout_index(index)
  File "/usr/lib/python3.8/site-packages/libqtile/core/manager.py", line 1059, in cmd_to_layout_index
    group.use_layout(index)
  File "/usr/lib/python3.8/site-packages/libqtile/group.py", line 129, in use_layout
    assert 0 <= index < len(self.layouts), "layout index out of bounds"
AssertionError: layout index out of bounds
```
And thanks for this awesome piece of work ;)

--- a/libqtile/backend/x11/xcore.py
+++ b/libqtile/backend/x11/xcore.py
@@ -535,7 +535,7 @@ class XCore(base.Core):
         if atoms["_NET_CURRENT_DESKTOP"] == opcode:
             index = data.data32[0]
             try:
-                self.qtile.cmd_to_layout_index(index)
+                self.qtile.groups[index].cmd_toscreen()
             except IndexError:
                 logger.info("Invalid Desktop Index: %s" % index)
 

